### ISSUE TITLE
Add System.Guid to TypeService.GetTsSimpleTypeName()

### ISSUE THREAD

**Issue by voronizer** [jburzynski/TypeGen#35]
It needs to be System.Guid to be resolved as String in TypeGen/src/TypeGen/TypeGen.Core/Business/TypeService.cs#GetTsSimpleTypeName()

**Comment by jburzynski** [comment]
Hi! Right now the solution would be to annotate properties with `TsTypeAttribute` like this:

```csharp
[TsType(TsType.String)]
public Guid GuidProperty { get; set; }
```

In the next release I'll try to provide some default mapping of common types (like Guid, DateTimeOffset etc.) to some "native" TS types. The next release will probably be out in a couple of days.

Cheers!

**Comment by voronizer** [comment]
Hi. Sounds great! Thanks!

BTW. I'm trying to keep main code clean, i.e. trying to not overload classes with attributes :). So, I created new console project in solution for creating ts files. Than imported main module into it and generate each file with `Generate(Type type)`.

**Comment by jburzynski** [comment]
Cool, it's nice to know that someone is also using the "programmatical" API :) I guess that's when it can be useful - when you have some scenario not covered by the TypeGen CLI. Hope it helps and works well! :)

**Comment by maxmalmgren** [comment]
Would be great if it was possible to inject the ITypeService instance or if there were some other way to affect the generated types. For example, for DateTime I'd like the TypeScript files to have the type string, and guid should map to string.

If I could provide ITypeService I wouldn't need to annotate all my source code. :)

**Comment by jburzynski** [comment]
It seeems like this (configurable default type mappings) can be a useful feature. If it's easy enough to implement, maybe it will be in the next release. I'll see what I can do and keep you updated.

**Comment by jburzynski** [comment]
@maxmalmgren I created a new topic for configurable default type mappings: #36 . Post any suggestions there. I'll also post any updates in #36 Thanks.

**Comment by jburzynski** [comment]
I've released version 2.0.1 (major version change is because of the changed syntax in the `TypeGen` nuget package - it's now the same as in the .NET CLI tool) and now Guid is mapped by default to string. The new version also has the functionality to define custom type mappings - I covered it [here](https://github.com/jburzynski/TypeGen/issues/36#issuecomment-444980195).
